packages/runtime-core/src/scheduler.ts

```diff
@@ -113,6 +113,13 @@ export function invalidateJob(job: SchedulerJob) {
   }
 }
 
+export function invalidatePostJob(job: SchedulerJob) {
+  const i = pendingPostFlushCbs.indexOf(job)
+  if (i > postFlushIndex) {
```

Comment:
I think there's a potential problem here because `postFlushIndex` is `0` when the queue isn't currently being processed, so this check can't remove the first job from the queue.

[Example](https://deploy-preview-9370--vue-sfc-playground.netlify.app/#eNqNUslOwzAQ/ZWRLy0SpAJuqEUs4gASi4CjLyGdJG4dO/LSFlX5d8Z2G1pUVdw8894sb/zW7LZts4VHdsXGtjCidWDR+faaK9G02jhYg8ESOiiNbmBA1EEPPX/f10JON1A22sShH5G4KrSy1K/WS5iELsMylxZPAlR6VTihFThdVRKHJ7DmCiI3W+TSI1U445Grbo9ukNY7yI69I308SkpIAwUOm1bmDikCGH9556jNTSFFMZ9wlsZzFlGAzxQG5ihRU9lW6eJMlFQVBnMGowiuw4XSVl2c3k9kp8xZOkIpqmxmtaIjx705K3TTConmtQ2qLGdXSVHAcin18inmwgVOt/mixmJ+ID+zq5Dj7C2sYRakpsdcbip0CX74eMEVvXuw0VMftB8B39Fq6cOOiXbn1ZTW3uHFbR+jH4SqPu3DyqGyW1HpCwG6yOeMnHF/RPrvupfZZayjD6Ur7hjrmFG1etZeOZz+sStXPTKkX5pcp4nBnlpiJnU1HNRo8HxA3uyiP/9XcLEtOGK5GulC+7bofgDHuizX)

Notice how clicking the button still logs `here1` but not `here2`.

A potential fix for that might be to have `postFlushIndex` set to `-1` when the queue isn't currently running.

In theory, `invalidateJob()` suffers from this same bug, but in practice `invalidateJob()` is only called from places where the scheduler queue is already running, so it never hits this particular edge case.

Reply:
Done